streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamFlatMap.java

```diff
@@ -41,6 +43,7 @@
         public void process(final Record<KIn, VIn> record) {
             final Iterable<? extends KeyValue<? extends KOut, ? extends VOut>> newKeyValues =
                 mapper.apply(record.key(), record.value());
+            Objects.requireNonNull(newKeyValues, String.format("KeyValueMapper can't return null from mapping the record: %s", record));
```

Comment:
Just a suggestion:
```suggestion
            Objects.requireNonNull(newKeyValues, "The provided KeyValueMapper returned null which is not allowed.");
```
BTW, we should not output records since they might contain sensitive data.